python/ray/tests/spark/test_databricks_hook.py

```diff
@@ -0,0 +1,86 @@
+import sys
+
+import mock
+import pytest
+import os
+import time
+import ray
+from pyspark.sql import SparkSession
+from ray.util.spark import setup_ray_cluster
+import ray.util.spark.databricks_hook
+
+
+pytestmark = pytest.mark.skipif(
+    not sys.platform.startswith("linux"),
+    reason="Ray on spark only supports running on Linux.",
+)
+ray.util.spark.databricks_hook._DATABRICKS_DEFAULT_TMP_DIR = "/tmp"
+
+
+class MockDbApiEntry:
+    def __init__(self):
+        self.idle_time = 0
+        self.registered_job_groups = []
+
+    def getIdleTimeMillisSinceLastNotebookExecution(self):
+        return (time.time() - self.idle_time) * 1000
+
+    def registerBackgroundSparkJobGroup(self, job_group_id):
+        self.registered_job_groups.append(job_group_id)
+
+
+class TestDatabricksHook:
+    @classmethod
+    def setup_class(cls):
+        os.environ["SPARK_WORKER_CORES"] = "2"
+        cls.spark = (
+            SparkSession.builder.master("local-cluster[1, 2, 1024]")
+            .config("spark.task.cpus", "1")
+            .config("spark.task.maxFailures", "1")
+            .config("spark.executorEnv.RAY_ON_SPARK_WORKER_CPU_CORES", "2")
+            .getOrCreate()
+        )
+
+    @classmethod
+    def teardown_class(cls):
+        time.sleep(10)  # Wait all background spark job canceled.
+        cls.spark.stop()
+        os.environ.pop("SPARK_WORKER_CORES")
+
+    def test_hook(self):
+        try:
+            db_api_entry = MockDbApiEntry()
+            with mock.patch(
+                "ray.util.spark.databricks_hook._get_db_api_entry",
+                return_value=db_api_entry,
+            ), mock.patch.dict(
+                os.environ,
+                {
+                    "DATABRICKS_RUNTIME_VERSION": "12.2",
+                    "DATABRICKS_RAY_ON_SPARK_AUTOSHUTDOWN_TIMEOUT_MINUTES": "2",
+                },
+            ):
+                setup_ray_cluster(
+                    num_worker_nodes=2,
+                    head_node_options={"include_dashboard": False},
+                )
+                cluster = ray.util.spark.cluster_init._active_ray_cluster
+                assert not cluster.is_shutdown
+                assert db_api_entry.registered_job_groups == [
+                    cluster.spark_job_group_id
+                ]
+                time.sleep(2.5)
```

Comment:
Why do we sleep for 2.5 seconds?

Reply:
Because we set `DATABRICKS_RAY_ON_SPARK_AUTOSHUTDOWN_TIMEOUT_MINUTES` to 2 minutes in test, and the patched `getIdleTimeMillisSinceLastNotebookExecution` api returns idle time that counts from the time that api_entry object is created. So after 2 minutes the cluster is shut down. 0.5 is a buffer considering the shutting down time

Reply:
`time.sleep(2.5)` only sleeps for 2.5 **SECONDS**, right?

Reply:
oh yes, my fault.